Latest Cryostat containers fail to deploy due to conf.d directory not existing

```
$ oc logs -c cryostat-sample -f pod/cryostat-sample-6dcbc876dd-j4l9g
Importing certificate /truststore/cryostat-sample-ca.crt ...
Certificate was added to keystore
[Storing /opt/cryostat.d/truststore.p12]
+ exec java -XX:+CrashOnOutOfMemoryError -Dcom.sun.management.jmxremote.autodiscovery=true -Dcom.sun.management.jmxremote.port=9091 -Dcom.sun.management.jmxremote.rmi.port=9091 -Djavax.net.ssl.trustStore=/opt/cryostat.d/truststore.p12 -Djavax.net.ssl.trustStorePassword=0E4EqrDMqMqIU1Ls7Eni5ro8jGF2ZTbD -Dcom.sun.management.jmxremote.authenticate=true -Dcom.sun.management.jmxremote.password.file=/tmp/jmxremote.password -Dcom.sun.management.jmxremote.access.file=/tmp/jmxremote.access -Dcom.sun.management.jmxremote.ssl.need.client.auth=true -Djavax.net.ssl.keyStore=/var/run/secrets/operator.cryostat.io/cryostat-sample-tls/keystore.p12 -Djavax.net.ssl.keyStorePassword=q-1vTltpR2cD1OI3cr5s -Dcom.sun.management.jmxremote.ssl=true -Dcom.sun.management.jmxremote.registry.ssl=true -cp '/app/resources:/app/classes:/app/libs/*:/*' io.cryostat.Cryostat
Jun 28, 2021 6:26:36 PM io.cryostat.core.log.Logger info
INFO: cryostat started.
Jun 28, 2021 6:26:36 PM io.cryostat.core.log.Logger info
INFO: Local config path set as /opt/cryostat.d/conf.d
Exception in thread "main" java.lang.RuntimeException: java.nio.file.NoSuchFileException: /opt/cryostat.d/conf.d/credentials
	at io.cryostat.configuration.ConfigurationModule.provideCredentialsManager(ConfigurationModule.java:92)
	at io.cryostat.configuration.ConfigurationModule_ProvideCredentialsManagerFactory.provideCredentialsManager(ConfigurationModule_ProvideCredentialsManagerFactory.java:51)
	at io.cryostat.configuration.ConfigurationModule_ProvideCredentialsManagerFactory.get(ConfigurationModule_ProvideCredentialsManagerFactory.java:40)
	at io.cryostat.configuration.ConfigurationModule_ProvideCredentialsManagerFactory.get(ConfigurationModule_ProvideCredentialsManagerFactory.java:12)
	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
	at io.cryostat.DaggerCryostat_Client.credentialsManager(DaggerCryostat_Client.java:617)
	at io.cryostat.Cryostat.main(Cryostat.java:70)
Caused by: java.nio.file.NoSuchFileException: /opt/cryostat.d/conf.d/credentials
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:389)
	at java.base/java.nio.file.Files.createDirectory(Files.java:690)
	at io.cryostat.configuration.ConfigurationModule.provideCredentialsManager(ConfigurationModule.java:82)
	... 6 more
```

--- a/internal/controllers/common/resource_definitions/resource_definitions.go
+++ b/internal/controllers/common/resource_definitions/resource_definitions.go
@@ -225,6 +225,10 @@ func NewPodForCR(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTags *I
 }
 
 func NewCoreContainer(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTag string, tls *TLSConfig) corev1.Container {
+	configPath := "/opt/cryostat.d/conf.d"
+	archivePath := "/opt/cryostat.d/recordings.d"
+	templatesPath := "/opt/cryostat.d/templates.d"
+	clientlibPath := "/opt/cryostat.d/clientlib.d"
 	envs := []corev1.EnvVar{
 		{
 			Name:  "CRYOSTAT_SSL_PROXIED",
@@ -243,8 +247,20 @@ func NewCoreContainer(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTa
 			Value: "9090",
 		},
 		{
+			Name:  "CRYOSTAT_CONFIG_PATH",
+			Value: configPath,
+		},
+		{
+			Name:  "CRYOSTAT_ARCHIVE_PATH",
+			Value: archivePath,
+		},
+		{
 			Name:  "CRYOSTAT_TEMPLATE_PATH",
-			Value: "/templates",
+			Value: templatesPath,
+		},
+		{
+			Name:  "CRYOSTAT_CLIENTLIB_PATH",
+			Value: clientlibPath,
 		},
 	}
 	if specs.CoreURL != nil {
@@ -286,13 +302,28 @@ func NewCoreContainer(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTa
 	mounts := []corev1.VolumeMount{
 		{
 			Name:      cr.Name,
-			MountPath: "flightrecordings",
+			MountPath: configPath,
+			SubPath:   "config",
+		},
+		{
+			Name:      cr.Name,
+			MountPath: archivePath,
 			SubPath:   "flightrecordings",
 		},
 		{
 			Name:      cr.Name,
-			MountPath: "templates",
+			MountPath: templatesPath,
 			SubPath:   "templates",
+		},
+		{
+			Name:      cr.Name,
+			MountPath: clientlibPath,
+			SubPath:   "clientlib",
+		},
+		{
+			Name:      cr.Name,
+			MountPath: "truststore",
+			SubPath:   "truststore",
 		},
 	}
 

--- a/internal/test/resources.go
+++ b/internal/test/resources.go
@@ -748,8 +748,20 @@ func NewCoreEnvironmentVariables(minimal bool, tls bool) []corev1.EnvVar {
 			Value: "cryostat-command.example.com",
 		},
 		{
+			Name:  "CRYOSTAT_CONFIG_PATH",
+			Value: "/opt/cryostat.d/conf.d",
+		},
+		{
+			Name:  "CRYOSTAT_ARCHIVE_PATH",
+			Value: "/opt/cryostat.d/recordings.d",
+		},
+		{
 			Name:  "CRYOSTAT_TEMPLATE_PATH",
-			Value: "/templates",
+			Value: "/opt/cryostat.d/templates.d",
+		},
+		{
+			Name:  "CRYOSTAT_CLIENTLIB_PATH",
+			Value: "/opt/cryostat.d/clientlib.d",
 		},
 	}
 	if !minimal {
@@ -848,14 +860,32 @@ func NewCoreVolumeMounts(tls bool) []corev1.VolumeMount {
 		{
 			Name:      "cryostat",
 			ReadOnly:  false,
-			MountPath: "flightrecordings",
+			MountPath: "/opt/cryostat.d/conf.d",
+			SubPath:   "config",
+		},
+		{
+			Name:      "cryostat",
+			ReadOnly:  false,
+			MountPath: "/opt/cryostat.d/recordings.d",
 			SubPath:   "flightrecordings",
 		},
 		{
 			Name:      "cryostat",
 			ReadOnly:  false,
-			MountPath: "templates",
+			MountPath: "/opt/cryostat.d/templates.d",
 			SubPath:   "templates",
+		},
+		{
+			Name:      "cryostat",
+			ReadOnly:  false,
+			MountPath: "/opt/cryostat.d/clientlib.d",
+			SubPath:   "clientlib",
+		},
+		{
+			Name:      "cryostat",
+			ReadOnly:  false,
+			MountPath: "truststore",
+			SubPath:   "truststore",
 		},
 	}
 	if tls {

